src/main.cpp

```diff
@@ -175,15 +182,21 @@ int main(int argc, const char* argv[]) {
         ->capture_default_str()
         ->ignore_case();
 
-    auto host_opt = app.add_subcommand("host", "HOST/CPU code backends")->ignore_case();
-    host_opt->add_flag("--c", c_backend, fmt::format("C/C++ backend ({})", c_backend))
+    auto simulator_opt =
+        app.add_subcommand("simulator", "NEURON/CoreNEURON code backends")->ignore_case();
+    simulator_opt->add_flag("--neuron", neuron_code, "Generate C++ code for NEURON")->ignore_case();
+    simulator_opt
+        ->add_flag("--coreneuron", coreneuron_code, "Generate C++ code for CoreNEURON (Default)")
```

Comment:
How would you expect to use a boolean flag?
Do you mean having something like:
```
nmodl file.mod --neuron
```
or
```
nmodl file.mod neuron --on
```
or
```
nmodl file.mod --neuron=on
```
?
Currently how it works is
```
nmodl file.mod simulator --neuron
```
to enable the `NEURON` codegen for example.

Reply:
I would say the easiest and most logical (to me) is to call nmodl like this when it should produce neuron code:

```
nmodl file.mod
```

or this when it should produce coreneuron code

```
nmodl file.mod --coreneuron
```

Reply:
Okay, I see what you mean. Would it make sense to keep the CoreNEURON code generation as default until we get the NEURON one ready? I'm considering this because otherwise we would need to add `--coreneuron` in `nrnivmodl`, etc and also have to write it ourselves when running nmodl.
Once NEURON code generation is in a good state we can change the default to NEURON.

Reply:
fair enough. Reason why i didn't suggest that is that I was worried that changing the logic might mess with people's assumptions down the line, but maybe this doesn't matter, especially if you ahve both a `--neuron` and a `--coreneuron` flag.